src/methods/selectKeys.ts

```diff
@@ -0,0 +1,41 @@
+import log from "../helpers/log.js"
+import { SimpleDataItem, Options, defaultOptions } from "../types.js"
+import showTable from "./showTable.js"
+
+export default function selectKeys(data: SimpleDataItem[], keys: string[], options: Options): SimpleDataItem[] {
+
+    const start = Date.now()
+
+    options = {
+        ...defaultOptions,
+        ...options
+    }
+
+    options.logs && log("\nselectKeys() " + keys)
+    options.logOptions && log("options:")
+    options.logOptions && log(options)
+
+    for (let key of keys) {
```

Comment:
Should we make that check in the `SimpleData` constructor instead?

Reply:
I think that we should make sure that the data has a correct format in the constructor. Once we have checked it, the data is only accessible from our methods so we do not have to make all these checks.

Reply:
Not sure you highlighted the right code. Did you?

In the constructor, we check if all items have the same keys. But in the function, we need to check if the objects have the key given by the user. Is that answering your question?